shell/renderer/api/electron_api_context_bridge.cc

```diff
@@ -810,13 +857,233 @@ bool OverrideGlobalPropertyFromIsolatedWorld(
+    v8::Isolate* isolate,
+    v8::Local<v8::Value> value,
+    v8::Local<v8::Context> target_context,
+    std::string& error_message) {
+  // Objects with prototype chains need to be protected and thus cloned.
+  // Primitive values are fine as is.
+  if (!value->IsObject()) {
+    return value;
+  }
+
+  v8::Local<v8::Object> object_value = value.As<v8::Object>();
+  v8::Local<v8::Context> creation_context =
+      object_value->GetCreationContextChecked();
+
+  // Value created in the same context are safe.
+  if (target_context == creation_context) {
+    return object_value;
+  }
+
+  // Attempt to clone value.
+  v8::MaybeLocal<v8::Value> maybe_result;
+  {
+    v8::TryCatch try_catch(isolate);
+    context_bridge::ObjectCache object_cache;
+    maybe_result =
+        PassValueToOtherContext(creation_context, target_context, object_value,
+                                creation_context->Global(), &object_cache,
+                                false, 0, BridgeErrorTarget::kSource);
+    if (try_catch.HasCaught()) {
+      v8::String::Utf8Value utf8(isolate, try_catch.Exception());
+      error_message = *utf8 ? *utf8 : "Unknown error cloning result";
+    }
+  }
+
+  return maybe_result;
+}
+
+// Serialize script to be executed in the given world.
+v8::Local<v8::Value> ExecuteInWorld(v8::Isolate* isolate,
+                                    const int world_id,
+                                    gin_helper::Arguments* args) {
+  // Get execution script argument
+  gin_helper::Dictionary exec_script;
+  if (args->Length() >= 1 && !args->GetNext(&exec_script)) {
+    gin_helper::ErrorThrower(args->isolate()).ThrowError("Invalid script");
+    return v8::Local<v8::Value>();
+  }
+
+  // Get "func" from execution script
+  v8::Local<v8::Function> func;
+  if (!exec_script.Get("func", &func)) {
+    gin_helper::ErrorThrower(isolate).ThrowError(
+        "Function 'func' is required in script");
+    return v8::Local<v8::Value>();
+  }
+
+  // Get optional "args" from execution script
+  v8::Local<v8::Array> args_array;
+  v8::Local<v8::Value> args_value;
+  if (exec_script.Get("args", &args_value)) {
+    if (!args_value->IsArray()) {
+      gin_helper::ErrorThrower(isolate).ThrowError("'args' must be an array");
+      return v8::Local<v8::Value>();
+    }
+    args_array = args_value.As<v8::Array>();
+  } else {
+    args_array = v8::Array::New(isolate);
+  }
+
+  // Serialize the function
+  v8::Local<v8::String> serialized_func;
+  if (!func->FunctionProtoToString(isolate->GetCurrentContext())
+           .ToLocal(&serialized_func)) {
+    gin_helper::ErrorThrower(isolate).ThrowError(
+        "Failed to serialize function");
+    return v8::Local<v8::Value>();
+  }
+  v8::String::Utf8Value serialized_func_utf8(isolate, serialized_func);
+  std::string func_str(*serialized_func_utf8, serialized_func_utf8.length());
+
+  v8::Local<v8::Context> source_context = isolate->GetCurrentContext();
```

Comment:
I think it's safer to get `exec_script`'s creation context 🤔

Reply:
I'll move this up to the top so it's not as risky if another context is entered.